contrib/neml2.mk

```diff
@@ -1,7 +1,77 @@
-ifeq ($(ENABLE_LIBTORCH),true)
-NEML2_DIR            ?= $(BLACKBEAR_DIR)/contrib/neml2
+# There are 2 influential conf vars.
+# They are controlled by the MOOSE configure script.
+#
+# ENABLE_LIBTORCH: True if we should compile with libTorch AND an installation
+#                  of libTorch is available. False otherwise.
+# LIBTORCH_DIR:    The path to the libTorch installation. NEML2 will use this
+#                  libTorch as the backend.
+#
+# There are 2 influential env vars.
+#
+# ENABLE_NEML2:  If true, BlackBear will attempt to compile with NEML2 at
+#                the NEML2_DIR. If false, BlackBear will not be compiled with
+#                NEML2 (even if the NEML2 submodule is init'ed or a valid
+#                NEML2_DIR is supplied). This variable defaults to true.
+# NEML2_DIR:     The path to a valid NEML2 checkout. This variable defaults to
+#                the NEML2 submodule. The user is responsible for setting this
+#                variable if a custom NEML2 should be used. The compilation
+#                will terminate with an error message if ENABLE_NEML2 is true
+#                AND a NEML2 checkout cannot be found.
 
-ifneq ($(wildcard $(NEML2_DIR)/CMakeLists.txt),)
+NEML2_DIR ?= $(BLACKBEAR_DIR)/contrib/neml2
+ifeq ($(wildcard $(NEML2_DIR)/CMakeLists.txt),)
+_ENABLE_NEML2_DEFAULT = false
```

Comment:
It seems like the true/false settings here are switched around, unless I'm missing something.

Reply:
I think this is saying if `$(NEML2_DIR)/CMakeLists.txt` doesn't exist, `_ENABLE_NEML2_DEFAULT` should be false. Is that not what you had in mind?

Reply:
This looks correct to me

Reply:
Oh, I thought it was saying that if that file _does_ exist, `_ENABLE_NEML2_DEFAULT` should be false. I was missing a subtle point in Makefile syntax.